lib/suse/connect/connection.rb

```diff
@@ -94,8 +97,33 @@ def json_request(method, path, params = {})
         raise SUSE::Connect::NetworkError, 'Check your network connection and try again. If it keeps failing, report a bug.'
       end
 
+      # Given an HTTP response, try to update the credentials file with the
+      # given 'System-Token' header.
+      def update_system_token!(response)
+        return unless System.credentials?
+
+        token = response.to_hash[SUSE::Toolkit::Utilities::SYSTEM_TOKEN_HEADER.downcase]&.first&.strip
```

Comment:
this throws `syntax error` for me

Reply:
Mhm really? What ruby version are you running this with?

Reply:
I'm guessing that from the safe navigation operator (`&.`)? That was introduced in ruby 2.3.0, and we are using at least 2.5.0. Other than that, I don't know what can throw such an error... Could you iterate more on this?

Reply:
Ok, it's failing on sle12sp1 and similar, with a ruby version older than `2.3.0`. Follow up PR coming :)